aiounifi/devices.py

```diff
@@ -345,3 +384,77 @@ def __getitem__(self, obj_id: int) -> Port:
     def __iter__(self) -> Iterator[Union[int, str]]:
         """Iterate over ports."""
         return iter(self.ports)
+
+
+class Outlet:
+    """Represents an outlet."""
+
+    def __init__(self, raw: dict) -> None:
+        """Initialize outlet."""
+        self.raw = raw
+
+    @property
+    def name(self) -> str:
+        """Name of outlet"""
+        return self.raw.get("name", "")
+
+    @property
+    def index(self) -> int:
+        """Outlet index."""
+        return self.raw.get("index")
+
+    @property
+    def has_relay(self) -> bool:
+        """Is the outlet controllable"""
+        return self.raw.get("has_relay", False)
+
+    @property
+    def has_metering(self) -> Optional[bool]:
+        """Is metering supported."""
+        return self.raw.get("has_metering")
+
+    @property
+    def relay_state(self) -> Optional[bool]:
+        """Is outlet power on."""
+        return self.raw.get("relay_state")
```

Comment:
All examples you've shared of relay_state shows it to always exist
```suggestion
    def relay_state(self) -> bool:
        """Is outlet power on."""
        return self.raw["relay_state"]
```

Reply:
I haven't seen an example of it yet, but I'd imagine that if has_relay is False, there might not be a relay_state property.  similarly, all of my outlets have has_metering set to False.  I don't see any properties about a metering status, or current or past power usage, but I would assume if the outlet had metering there would be additional properties related to metering.

Reply:
Lets do it strict at least initially, easier to catch if some new device change the way it reports